hoomd/jit/GPUEvalFactory.h

```diff
@@ -101,16 +74,22 @@ class GPUEvalFactory
          */
         unsigned int getKernelMaxThreads(unsigned int idev, unsigned int eval_threads, unsigned int launch_bounds)
             {
-            assert(m_kernel_ptr[std::make_pair(eval_threads,launch_bounds)].size() > idev);
             int max_threads = 0;
-            CUresult custatus = cuFuncGetAttribute(&max_threads, CU_FUNC_ATTRIBUTE_MAX_THREADS_PER_BLOCK,
-                m_kernel_ptr[std::make_pair(eval_threads,launch_bounds)][idev]);
+
+            #ifdef __HIP_PLATFORM_NVCC__
+            cudaSetDevice(m_exec_conf->getGPUIds()[idev]);
+            CUresult custatus = cuFuncGetAttribute(&max_threads,
+                CU_FUNC_ATTRIBUTE_MAX_THREADS_PER_BLOCK,
+                m_program->kernel(m_kernel_name).instantiate(eval_threads, launch_bounds));
             char *error;
             if (custatus != CUDA_SUCCESS)
                 {
                 cuGetErrorString(custatus, const_cast<const char **>(&error));
-                throw std::runtime_error("cufuncGetAttribute: "+std::string(error));
+                throw std::runtime_error("cuFuncGetAttribute: "+std::string(error));
                 }
+            cudaSetDevice(m_exec_conf->getGPUIds()[0]);
+            #endif
```

Comment:
Is there no corresponding logic for HIP? Do you not have to specify the device, or does it not support multi-gpu?

Reply:
I suggest we implement RTC with HIP later, if it is at all possible.

Reply:
HIP doesn't support managed memory/multi-GPU just yet